src/components/ShareMyLocationButton.js

```diff
@@ -49,7 +50,7 @@ const ShareMyLocationButton = () => {
     }
     setUserLocationShared(false);
     sharingLocation.current = false;
-    setNewCenterMap(defaultCenterMap);
+    setNewCenterMap({ ...defaultCenterMap, isDragged: true });
```

Comment:
What do you think if you use two separate, independent states `defaultCenterMap` & `isDragged` 😄 ?

Reply:
@alxmcr, please see the explanation above

Reply:
I leave some comments ^^^. Also, you can avoid destructing `defaultCenterMap` object if you use a new, separate state 😄 .